src/pretalx/submission/models/submission.py

```diff
@@ -288,39 +299,70 @@ def _set_state(self, new_state, force=False, person=None):
                 )
             )
 
-    def make_submitted(self, person=None, force=False, orga=False):
-        self._set_state(SubmissionStates.SUBMITTED, force, person=person)
+    def update_talk_slots(self, slot_count_old=None):
         from pretalx.schedule.models import TalkSlot
+        # logging.debug('update_talk_slots')
+        # debug: print all now available Talks
+        # qs_temp = TalkSlot.objects.filter(
+        #     submission=self,
+        #     schedule=self.event.wip_schedule,
+        # )
+        # logging.debug('TalkSlots old: {}'.format(
+        #     [i.slot_index for i in qs_temp]))
+        # logging.debug('self.state: {}'.format(self.state))
+        if self.state == SubmissionStates.ACCEPTED or \
+                self.state == SubmissionStates.CONFIRMED:
+            # logging.debug('state is ACCEPTED or CONFIRMED')
+            if slot_count_old is not None:
+                # logging.debug(
+                #     'slot_count_old: {}'.format(slot_count_old))
+                # logging.debug(
+                #     'self.slot_count {}'.format(self.slot_count))
+                range_to_delete = list(range(
+                    self.slot_count, slot_count_old))
+                # logging.debug('range_to_delete: {}'.format(range_to_delete))
+                qs_delete = TalkSlot.objects.filter(
+                    submission=self,
+                    schedule=self.event.wip_schedule,
+                    slot_index__in=range_to_delete,
```

Comment:
If we remove the `slot_index` field, we could just prefer to delete unscheduled talk slots first, and scheduled talk slots later.

Reply:
done